### ISSUE TITLE
On relation_broken, `IngressPerUnitRequirer` still displays the URL no longer available

### ISSUE THREAD

**Issue by mmanciop** [canonical/traefik-k8s-operator#32]
### Bug Description

When removing an `ingress_per_unit` relation, the `IngressPerUnitRequirer.url` property return the last ingress URL.

### To Reproduce

1. juju deploy traefik-k8s --channel=edge --trust
2. juju deploy prometheus-k8s --channel=edge --trust
3. juju add-relation traefik-k8s prometheus-k8s:ingress
4. Ensure that traefik is in Active status (e.g., setting up `microk8s enable meltallb`)
5. juju remove-relation traefik-k8s prometheus-k8s:ingress
6. microk8s.kubectl exec prometheus-k8s-0 -n cos -c prometheus -- ps aux

### Environment

My nightmares

### Relevant log output

```shell
Adding `print(f"Update grafana source: {self._external_url}")`in the constructor of Prometheus:

unit-prometheus-k8s-0: 13:43:38 DEBUG unit.prometheus-k8s/0.ingress-relation-broken External URL: http://192.168.2.10:80/cos-prometheus-k8s-0
unit-prometheus-k8s-0: 13:43:38 INFO unit.prometheus-k8s/0.juju-log ingress:7: Prometheus configuration reloaded
unit-prometheus-k8s-0: 13:43:38 DEBUG unit.prometheus-k8s/0.juju-log ingress:7: Setting Grafana data sources: {'model': 'cos', 'model_uuid': '3f8cf4e2-dca6-4bef-8e4f-4fa67728d12b', 'application': 'prometheus-k8s', 'type': 'prometheus'}
unit-prometheus-k8s-0: 13:43:38 DEBUG unit.prometheus-k8s/0.ingress-relation-broken Update grafana source: http://192.168.2.10:80/cos-prometheus-k8s-0
unit-prometheus-k8s-0: 13:43:39 INFO juju.worker.uniter.operation ran "ingress-relation-broken" hook (via hook dispatching script: dispatch)
```

### Additional context

My guess is that the relation data bags are still accessible during the `relation_broken` event, and that leads to spurious results.

**Comment by mmanciop** [comment]
@simskij the sad part is that you and I must have seen this bug play out a million times, and blamed it on the replan bug, as it too gets solved by a pod churn :-(

**Comment by mmanciop** [comment]
The problem seems to be that the `relation_broken` event is not sufficient to mark the relation as "not ready".

**Comment by mmanciop** [comment]
Fixed via 35594c05f119ae27b1163048a5c8ff49588aa7d8

**Comment by mmanciop** [comment]
@Abuelodelanada you will also need the new version of the library in Loki
